CMakeLists.txt

```diff
@@ -80,15 +80,18 @@ if( USE_PACKAGE_COMPILER )
     # Define PackageCompiler.jl output file
     set( PC_LIBTRIXI_SO ${CMAKE_BINARY_DIR}/prefix-pc/lib/libtrixi.so )
 
-    # Add a library target (libtrixi), but only for Fortran module
+    # Add a library target (libtrixi)
+    # - for Fortran module (therefore OBJECT library)
+    # - for generating a rule to build libtrixi.so
     add_library( ${PROJECT_NAME} OBJECT
         src/api.f90
+        ${PC_LIBTRIXI_SO}
```

Comment:
Why is this necessary?

Reply:
Good question!

Short answer: it does not work otherwise

Long answer:

> This [add_custom_command] defines a command to generate specified OUTPUT file(s). A target created in the same directory (CMakeLists.txt file) that specifies any output of the custom command as a source file is given a rule to generate the file using the command at build time.

So, if I understand this correctly, the output (libtrixi.so) has to appear as a source file in order to create to a build rule for it.

Reply:
You might need to add an `add_custom_target` (https://dev.to/iblancasa/learning-cmake-3-understanding-addcustomcommand-and-addcustomtarget-43gp) or otherwise use a full path in `target_link_libraries` (https://stackoverflow.com/a/44084377/1329844). Not sure if these apply here, but maybe it helps.

Maybe instead of adding it as a source, you could also add the `${PC_LIBTRIXI_SO}` through `target_link_libraries` (https://cmake.org/cmake/help/latest/command/target_link_libraries.html#command:target_link_libraries)?

I am not sure what the correct setup is for this, maybe we have to ask on the CMake Discourse forum. The challenge here is imho that usually CMake expects to build the library itself, or to use an existing library somewhere, but not this mixture that we have 🤷‍♂️

Reply:
Thanks for the pointers! I will try `add_custom_target`.

Just for explanation: `${PC_LIBTRIXI_SO}` actually is a full path, and `target_link_libraries` is already used (and apparently required).